Python/syntax_test_python.py

```diff
@@ -1350,6 +1350,74 @@ class Starship:
 #        ^ punctuation.separator.annotation.variable.python
 #                                   ^ keyword.operator.assignment
 
+# Type comments - type: ignore must be by itself.
+
+primes = 5  # type: ignore # type: not-a-type-comment
+#           ^ comment.line.type-hint punctuation.definition.comment.python comment.line.number-sign.python
+#             ^^^^ comment.line.type-hint keyword.other.annotation.type-comment
+#                 ^ comment.line.type-hint punctuation.separator.annotation.type-comment
+#                   ^^^^^^ comment.line.type-hint keyword.other.annotation.type-comment.ignore
+#                          ^ comment.line.type-hint comment.line.number-sign.python punctuation.definition.comment.python
+#                            ^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.type-hint comment.line.number-sign.python
+
+# This is not a type-ignore comment.
+value = 3  # type: ignore_class_starting_with_ignore
+#          ^ comment.line.type-hint.python punctuation.definition.comment.python comment.line.number-sign.python
+#            ^^^^ comment.line.type-hint.python keyword.other.annotation.type-comment.python
+#                ^ comment.line.type-hint.python punctuation.separator.annotation.type-comment.python
+#                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.type-hint.python support.class.python
+
+# Mypy ignore extension
+value = 3  # type: ignore[error-code-1, 4noint][error]
+#                  ^^^^^^ comment.line.type-hint.python keyword.other.annotation.type-comment.ignore
+#                        ^ comment.line.type-hint.python comment.line.type-hint.python punctuation.section.brackets.begin.python
+#                         ^^^^^^^^^^^^ comment.line.type-hint.python comment.line.type-hint.python.mypy-errorcode
+#                                     ^ comment.line.type-hint.python punctuation.separator.sequence.python.mypy-errorcode
+#                                       ^ invalid.illegal.python
+#                                             ^ comment.line.type-hint.python punctuation.section.brackets.end.python
+#                                              ^ invalid.illegal.python
+
+# With/for allow commas
+for a, b in lst: # type: str, int
+#                           ^ comment.line.type-hint.python punctuation.separator.sequence.python
+
+
+primes = 5  # type: List[Dict[*property:str, ...]], bool # comment
```

Comment:
@FichteFoll What do you think about using `clear_scopes` during the type definition to remove the `comment` scope and make it more readable? I wouldn't remove it from the `type: ` part, but maybe the `List[Dict[*property:str, ...]], bool` part?

Reply:
For color schemes that have a background color for comments, I think it will look weird to remove the comment scope for something that clearly is a comment syntactically. We don't do this in JavaDoc and similar either.

For those that only define a foreground color (the majority), it may look weird to the punctuation characters in the comment color. Additionally, it may confuse plugins that rely on scope selectors, even though the only one that comes to mind is SublimeLinter-annotations and that's not affected.

I guess with the latter much more wide-spread, we could try removing the `comment` scope, add a meta scope for accuracy and see how that goes.

Reply:
Alright, seems there are enough significant side effects to not bother.